generated/language/src/main/java/com/google/cloud/language/v1/spring/LanguageServiceSpringAutoConfiguration.java

```diff
@@ -0,0 +1,471 @@
+ *   <li>Retries are configured for idempotent methods but not for non-idempotent methods.
+ * </ul>
+ */
+@Generated("by gapic-generator-java")
+@BetaApi("Autogenerated Spring autoconfiguration is not yet stable")
+@AutoConfiguration
+@ConditionalOnClass(LanguageServiceClient.class)
+@ConditionalOnProperty(
+    value = "com.google.cloud.language.v1.spring.auto.language-service.enabled",
+    matchIfMissing = false)
+@EnableConfigurationProperties({LanguageServiceSpringProperties.class, GlobalProperties.class})
+public class LanguageServiceSpringAutoConfiguration {
+  private final LanguageServiceSpringProperties clientProperties;
+  private final GlobalProperties globalProperties;
+  private static final Log LOGGER = LogFactory.getLog(LanguageServiceSpringAutoConfiguration.class);
+
+  protected LanguageServiceSpringAutoConfiguration(
+      LanguageServiceSpringProperties clientProperties, GlobalProperties globalProperties) {
+    this.clientProperties = clientProperties;
+    this.globalProperties = globalProperties;
+  }
+
+  /**
+   * Obtains the default credentials provider. The used key will be obtained from Spring Boot
+   * configuration data files.
+   */
+  @Bean
+  @ConditionalOnMissingBean
+  public CredentialsProvider languageServiceCredentials() throws IOException {
+    if (this.clientProperties.getCredentials().hasKey()) {
+      return ((CredentialsProvider) new DefaultCredentialsProvider(this.clientProperties));
+    }
+    return ((CredentialsProvider) new DefaultCredentialsProvider(this.globalProperties));
+  }
+
+  /**
+   * Returns the default channel provider. The default is gRPC and will default to it unless the
+   * useRest option is provided to use HTTP transport instead
+   */
+  @Bean
+  @ConditionalOnMissingBean
+  public TransportChannelProvider defaultLanguageServiceTransportChannelProvider() {
+    return LanguageServiceSettings.defaultTransportChannelProvider();
+  }
+
+  /**
+   * Provides a LanguageServiceClient bean configured to use the default credentials provider
+   * (obtained with languageServiceCredentials()) and its default transport channel provider
+   * (defaultLanguageServiceTransportChannelProvider()). It also configures the quota project ID if
+   * provided. It will configure an executor provider in case there is more than one thread
+   * configured in the client
+   *
+   * <p>Individual retry settings are configured as well. It will use the relevant client library's
+   * default retry settings when they are not specified in LanguageServiceSpringProperties.
+   */
+  @Bean
+  @ConditionalOnMissingBean
+  public LanguageServiceClient languageServiceClient(
+      @Qualifier("languageServiceCredentials") CredentialsProvider credentialsProvider,
+      @Qualifier("defaultLanguageServiceTransportChannelProvider")
+          TransportChannelProvider defaultTransportChannelProvider)
+      throws IOException {
+    LanguageServiceSettings.Builder clientSettingsBuilder =
```

Comment:
Why not make `LanguageServiceSettings` its own bean so that it can be overwritten, and then have a separate simple method that takes in `LanguageServiceSettings` and creates the `LanguageServiceClient`?

Reply:
This is a good idea. I'll add to our todo list.